tests/local/atomics.wast

```diff
@@ -8,18 +8,20 @@
   (func (result i32)
     i32.const 0
     i32.const 0
-    memory.atomic.notify 0 offset=0
+    memory.atomic.notify offset=0
   )
 )
 
-(assert_invalid
-  (module
-    (memory 1)
-    (func (result i32)
-      i32.const 0
-      i32.const 0
-      i64.const 0
-      memory.atomic.wait32 1 offset=0
-    )
-  )
-  "unknown memory 1")
+;; Disabled because memory indices are not currently supported
```

Comment:
To confirm, is the intention that you'd write:

```wasm
v128.load8_lane offset=3 3 
```

for loading lane 3 at offset 3? As opposed to

```wasm
v128.load8_lane 3 offset=3  
```

(I figured the land would want to be prioritized here?)

Reply:
I thought this would be the latter, that is, since otherwise this breaks the multi-memory support for all instructions in `wast` I think which I think may be a bit too breaking for now. Changing from an index/identifier to `(memory $m)` should resolve the ambiguity there, though, and could be implemented for now. That'd be parsed as `Option<ItemRef<kw::memory>>` in `wast`

Reply:
Yes, confirmed.  According to the SIMD spec, the memory argument comes before the lane argument in the binary form, and arguably this is best for the text form too.  (The SIMD spec, alas, uses a C++-oriented form and is not a ton of help, but the order of operands is memory arg before lane.)

I'm not sure what you mean by the comment that it breaks all multi-memory support.  With the current code, which does not parse any memory index, there is no multi-memory support, hence all test cases that use a memory index are commented out.

Andreas also thought (memory $m) would be the correct syntax.  I can look into making that work, and will re-enable tests if and when it works.

Reply:
Ok makes sense. If you'd like I can land the change to `(memory ..)` before this PR since ideally this PR wouldn't be affected by multi-memory at all.

Reply:
Your call.  Since my patch now seems to do what it needs to do, I'm fine either way.